src/OmniSharp.Host/MSBuild/Discovery/MSBuildLocator.cs

```diff
@@ -70,6 +70,17 @@ public void RegisterInstance(MSBuildInstance instance)
 
             RegisteredInstance = instance ?? throw new ArgumentNullException(nameof(instance));
 
+            if (instance.DiscoveryType == DiscoveryType.StandAlone)
+            {
+                // MSBuild began relying on the Microsoft.IO.Redist library which does not work
+                // on MacOS or Linux platforms. Disabling the 17.0 feature wave opts us in to the
+                // earlier behavior of using System.IO.
+
+                // This issue only affects the MSBuild tools shipped with O# as Mono does not ship
+                // with a version of MSBuild that includes this change.
+                Environment.SetEnvironmentVariable("MSBUILDDISABLEFEATURESFROMVERSION", "17.0");
```

Comment:
this is some dark magic man

I think we need to expedite getting rid of the standalone msbuild feature as we discussed.

Reply:
I agree. I opened an issue and it should get resolved for 17.2 (https://github.com/dotnet/msbuild/issues/7335), but not shipping is best path forward.